### PR TITLE
task 2: Components

### PR DESCRIPTION
**Pull request by serhii-yaremchuk-epam**

### CODE REVIEW

File: my-app/src/app/core/header/breadcrumbs/breadcrumbs.component.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  selector: 'cp-breadcrumbs',
+  templateUrl: './breadcrumbs.component.html',
+  styleUrls: ['./breadcrumbs.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class BreadcrumbsComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit(): void {
```
Comment (vitaliy-bobrov) [inline my-app/src/app/core/header/breadcrumbs/breadcrumbs.component.ts]: [nit] remove if not used

File: my-app/src/app/features/courses-page/courses-item/courses-item.component.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+@Component({
+  selector: 'cp-courses-item',
+  templateUrl: './courses-item.component.html',
+  styleUrls: ['./courses-item.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class CoursesItemComponent implements OnInit, OnChanges, DoCheck, AfterContentInit, AfterContentChecked, AfterViewInit, AfterViewChecked, OnDestroy {
+  @Input() course!: Course;
+  @Output() delete: EventEmitter<Course> = new EventEmitter<Course>();
```
Comment (vitaliy-bobrov) [inline my-app/src/app/features/courses-page/courses-item/courses-item.component.ts]: `@Output() delete = new EventEmitter<Course>();` will be enough ;)

File: my-app/src/app/features/courses-page/courses-item/courses-item.component.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import {Course} from '../../../shared/models/course.model';
+
+@Component({
+  selector: 'cp-courses-item',
+  templateUrl: './courses-item.component.html',
+  styleUrls: ['./courses-item.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class CoursesItemComponent implements OnInit, OnChanges, DoCheck, AfterContentInit, AfterContentChecked, AfterViewInit, AfterViewChecked, OnDestroy {
+  @Input() course!: Course;
```
Comment (vitaliy-bobrov) [inline my-app/src/app/features/courses-page/courses-item/courses-item.component.ts:22]: would be nice to have a blank line between fields

File: my-app/src/app/features/courses-page/courses-page.component.html [hunk 4]
```diff
@@ -1 +1,27 @@
-<p>courses-page works!</p>
+<div class="courses-page">
+  <div class="courses-page__header justify--space-between">
+    <div class="courses-page-search">
+      <input [(ngModel)]="searchQuery" c
```
Comment (vitaliy-bobrov) [inline my-app/src/app/features/courses-page/courses-page.component.html]: `c` leftover

File: my-app/src/app/features/courses-page/courses-page.component.ts [hunk 5]
```diff
@@ -1,15 +1,62 @@
 import { Component, OnInit } from '@angular/core';
+import {Course} from '../../shared/models/course.model';
 
 @Component({
   selector: 'cp-courses-page',
   templateUrl: './courses-page.component.html',
   styleUrls: ['./courses-page.component.scss']
 })
 export class CoursesPageComponent implements OnInit {
+  public searchQuery = '';
```
Comment (vitaliy-bobrov) [inline my-app/src/app/features/courses-page/courses-page.component.ts]: `public` could be safely omitted there

File: my-app/src/app/shared/pipes/duration.pipe.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'duration'
+})
+export class DurationPipe implements PipeTransform {
+
+  transform(value: number, ...args: unknown[]): unknown {
```
Comment (vitaliy-bobrov) [inline my-app/src/app/shared/pipes/duration.pipe.ts]: return type should be `string`

File: my-app/src/app/shared/pipes/duration.pipe.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'duration'
```
Comment (vitaliy-bobrov) [inline my-app/src/app/shared/pipes/duration.pipe.ts:4]: consider to add `pure: true`, see https://angular.io/api/core/Pipe

File: my-app/src/scss/fonts.scss [hunk 8]
```diff
@@ -0,0 +1,13 @@
+@font-face {
+  font-family: 'SourceSansPro-Bold';
+  src: url('../assets/fonts/Source_Sans_Pro/SourceSansPro-Bold.ttf') format('truetype');
+  font-weight: normal;
```
Comment (vitaliy-bobrov) [inline my-app/src/scss/fonts.scss:4]: you can declare fonts to be the same font-family but with different font-weight

**Review by vitaliy-bobrov — APPROVED** [review]

File: my-app/src/app/shared/pipes/duration.pipe.ts (hunk 7 above)
Comment (serhii-yaremchuk-epam) [reply my-app/src/app/shared/pipes/duration.pipe.ts:4]: The documentation says "Pipes are pure by default."

Comment (vitaliy-bobrov) [reply my-app/src/app/shared/pipes/duration.pipe.ts:4]: sorry, my bad, they finally applied the default to `true`

**Pull request merged**